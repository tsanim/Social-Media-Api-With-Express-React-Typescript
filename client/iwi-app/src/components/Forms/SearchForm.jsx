import React from 'react';
import useForms from '../../hooks/useForms';
import PropTypes from 'prop-types';

function SearchForm({ searchHandler }) {
    const { handleSubmit, handleChangeInput, inputs } = useForms((e) => {
        searchHandler(inputs);
    })

    return (
        <form id="searchForm" onSubmit={handleSubmit}>
            <input
                type="search"
                placeholder="Search..."
                name="searchText"
                value={inputs.searchText || ''}
                onChange={handleChangeInput}
            />
        </form>
    )
}

SearchForm.propTypes = {
    searchHandler: PropTypes.func,
}

export default SearchForm;
