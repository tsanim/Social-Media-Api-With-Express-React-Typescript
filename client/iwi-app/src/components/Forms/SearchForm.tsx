import React from 'react';
import useForms from '../../hooks/useForms';
import PropTypes from 'prop-types';
import SearchFormProps from '../../interfaces/Components/Forms/SearchFormProps.interface';
import SearchData from '../../interfaces/SearchData.interface';

function SearchForm({ searchHandler }: SearchFormProps) {
    const { handleSubmit, handleChangeInput, inputs } = useForms(() => {
        searchHandler(inputs as SearchData);
    })

    return (
        <form id="searchForm" onSubmit={handleSubmit}>
            <input
                type="search"
                placeholder="Search..."
                name="searchText"
                value={inputs.searchText || ''}
                onChange={handleChangeInput}
            />
        </form>
    )
}

SearchForm.propTypes = {
    searchHandler: PropTypes.func,
}

export default SearchForm;
