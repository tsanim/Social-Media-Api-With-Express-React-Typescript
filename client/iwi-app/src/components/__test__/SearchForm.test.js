import React from 'react';
import { shallow } from 'enzyme';

import SearchForm from '../Forms/SearchForm';

function searchFormComponentSetup() {
    const props = {
        searchHandler: jest.fn(),
    }

    const enzymeWrapper = shallow(<SearchForm {...props} />)

    return {
        props,
        enzymeWrapper
    }
}


describe('SearchForm Component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = searchFormComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render container form tag with id "searchForm"', () => {
        const { enzymeWrapper } = searchFormComponentSetup();
        
        expect(enzymeWrapper.find('form#searchForm')).toHaveLength(1);
    });

    it('should render input with type search ', () => {
        const { enzymeWrapper } = searchFormComponentSetup();
        
        expect(enzymeWrapper.find('input').prop('type')).toEqual('search');
    });

    it('should render input with name "searchText"', () => {
        const { enzymeWrapper } = searchFormComponentSetup();
        
        expect(enzymeWrapper.find('input').prop('name')).toEqual('searchText');
    });

    it('should render input with onChange event', () => {
        const { enzymeWrapper } = searchFormComponentSetup();
        
        expect(enzymeWrapper.find('input').prop('onChange')).toBeDefined();
    });

    it('should test if user text is echoed', () => {
        const { enzymeWrapper } = searchFormComponentSetup();

        const mockedEvent = {
            target: { value: 'hello', name: 'searchText' },
            persist: jest.fn()
        };

        enzymeWrapper.find('input').simulate('change', mockedEvent);

        expect(enzymeWrapper.find('input').props().value).toEqual('hello');
    });

    it('should test if submit event when form is submitted', () => {
        const { enzymeWrapper, props } = searchFormComponentSetup();

        const mockedEvent = {
            preventDefault: () => {},
        };

        enzymeWrapper.find('form').simulate('submit', mockedEvent);

        expect(props.searchHandler).toHaveBeenCalledTimes(1);
    });
});
