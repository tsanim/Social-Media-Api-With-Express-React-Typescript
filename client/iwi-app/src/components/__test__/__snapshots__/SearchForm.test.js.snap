// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Component should render initial layout 1`] = `
Array [
  <form
    id="searchForm"
    onSubmit={[Function]}
  >
    <input
      name="searchText"
      onChange={[Function]}
      placeholder="Search..."
      type="search"
      value=""
    />
  </form>,
]
`;
